contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -648,15 +648,20 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 	List *path_oids = fetch_search_path(false);
 	char *old_search_path = flatten_search_path(path_oids);
 	char *new_search_path;
+	estate->db_name = NULL;
```

Comment:
does this code path also cover sp_execute?

Reply:
No, it does not. This behaviour is not expected for T-SQL batches and for EXEC() and sp_executesql.
The schema name resolution is only driven by the current user's default schema(which is already implemented).